allensdk/api/cloud_cache/cloud_cache.py

```diff
@@ -1107,3 +1220,82 @@ def _download_manifest(self, manifest_name: str):
 
     def _download_file(self, file_attributes: CacheFileAttributes) -> bool:
         raise NotImplementedError()
+
+
+class StaticLocalCache(BasicLocalCache):
+    """A class to handle accessing data that has already been downloaded
+    locally and whose directory structure and/or contained files are not
+    expected to be changed in any way. Does NOT support multiple manifest
+    versions for a given dataset.
+
+    Example intended use case:
+        Calling
+        VisualBehaviorOphysProjectCache.from_local_cache(use_static_cache=True)
+        where the cache directory is a mounted S3 public bucket.
+
+    Parameters
+    ----------
+    cache_dir: str or pathlib.Path
+        Path to the directory where data will be stored on the local system
+
+    project_name: str
+        the name of the project this cache is supposed to access. This will
+        be the root directory for all files stored in the bucket.
+
+    ui_class_name: Optional[str]
+        Name of the class users are actually using to maniuplate this
+        functionality (used to populate helpful error messages)
+    """
+
+    def __init__(self, cache_dir, project_name, ui_class_name=None):
+        super().__init__(cache_dir=cache_dir, project_name=project_name,
+                         ui_class_name=ui_class_name)
+
+    def _list_all_manifests(self) -> list:
+        """
+        Return a list of all of the file names of the manifests associated
+        with this dataset. For the StaticLocalCache only return only the
+        latest manifest.
+        """
+        manifest_dir = os.path.join(
+            self._cache_dir, self.project_name, "manifests"
+        )
+
+        output = [x for x in os.listdir(manifest_dir)
+                  if re.fullmatch(".*_manifest_v.*.json", x)]
+
+        return [self._find_latest_file(output)]
+
+    def list_all_downloaded_manifests(self) -> list:
+        """
+        Return a list of all of the manifest files for this dataset.
+        For the StaticLocalCache, this will only be the latest manifest.
+        """
+        return self._list_all_manifests()
+
+    def load_last_manifest(self):
+        """For the StaticLocalCache always load the latest manifest."""
+        self.load_manifest(self.latest_manifest_file)
+
+    def load_manifest(self, manifest_name: str):
```

Comment:
@danielsf The `BasicLocalCache` `_load_manifest()` method already checks that the `manifest_name` exists in `self.manifest_file_names` (self.manifest_file_names itself is determined by the `_list_all_manifests()` method which is overloaded in the `StaticLocalCache` to only ever return the manifest with the highest version.

Reply:
You're right. That is fine. Do you mind adding a unit test to `test_static_local_cache` to make sure that it only ever returns the latest manifest file (in case we accidentally break this behavior in the future)?

Reply:
Tests have been added.